src/main/java/uk/gov/companieshouse/charges/data/service/ChargesService.java

```diff
@@ -115,4 +124,55 @@ public Optional<ChargeApi> getChargeDetails(final String companyNumber, final St
         return chargeDetails;
     }
 
+    /**
+     * Find charges for company number.
+     *
+     * @param companyNumber company Number.
+     * @return charges.
+     */
+    public Optional<ChargesApi> findCharges(final String companyNumber, final Pageable pageable) {
+        logger.debug(String.format("Started : findCharges for Company Number %s ",
+                companyNumber
+        ));
+
+        List<ChargesDocument> charges =
+                this.chargesRepository.findCharges(companyNumber, pageable).getContent();
+        if (charges.isEmpty()) {
+            logger.error(
+                    String.format(
+                            "Finished: findCharges No charges found for company %s ",
+                            companyNumber));
+            return Optional.empty();
+        }
+
+        Optional<MetricsApi> companyMetrics =
+                companyMetricsApiService.getCompanyMetrics(companyNumber);
+
+        if (companyMetrics.isEmpty()) {
+            logger.error(
+                    String.format(
+                            "Finished: findCharges No company metrics data found for company %s ",
+                            companyNumber));
+            return Optional.empty();
+        }
+        var result = companyMetrics.map(metrics -> {
+            var chargesApi = new ChargesApi();
```

Comment:
Can we please add a test for this class?

Reply:
Test added